src/cheat.rs

```diff
@@ -161,3 +183,49 @@ pub fn read_all(config: &Config, stdin: &mut std::process::ChildStdin) -> HashMa
 
     variables
 }
+
+#[cfg(test)]
```

Comment:
Could you please move the tests to the /tests folder?

Reply:
sure, can do, but being able of putting the tests in the same file then the code, is one of the best features of rust.

Reply:
I wanted to point out, I have to make functions public to make the tests work, which don't need to be public.

Reply:
https://doc.rust-lang.org/stable/book/ch11-03-test-organization.html#testing-private-functions
I follow these recommendation/descriptions.

Reply:
Please let me know if I really should move the tests to `/tests` and make all the functions tested public.

Reply:
Wow! Didn't know this was idiomatic in Rust. It seems very counter-intuitive from my experience hahaha

But there must be good reasons for that (I haven't read the documents yet, sorry) so ignore my previous comment :+1:

Reply:
:+1: 
That was actually one of the first features that draw my attention to rust, because tests are one of the best ways to document code.